### DISCUSSION TITLE
get/set methods for hd.state

### DISCUSSION THREAD

**Discussion by tkriplean** [hyperdiv/hyperdiv#51] [Ideas]
I'd like to be able to do things like:

```
def update_endpoint(endpoint, state):
    current_val = state.get(f"clip_{endpoint}"))
    inp = hd.text_input(value=current_val)
    if inp.changed:
        state.set(f"clip_{endpoint}", inp.value)

state = hd.state(clip_start=None, clip_end=None)

update_endpoint("start", state)
update_endpoint("end", state)
```

I see that hd.state already have __getattribute__ and __setattr__. Can those be exposed?

For my current use case, there is an easy workaround, but I can imagine cases where it wouldn't apply.

Sorry if I'm missing an easy way to do this!

**Comment by mariusnita** [comment]
You can use `getattr(state, f"clip_{endpoint}")` and `setattr(state, f"clip_{endpoint}, inp.value)`.

https://docs.python.org/3/library/functions.html#getattr
https://docs.python.org/3/library/functions.html#setattr

**Comment by tkriplean** [comment]
Thanks Marius. Sorry I wasn't aware of that approach.
